kore/src/Kore/Unification/Procedure.hs

```diff
@@ -52,10 +52,11 @@ unificationProcedure
         , FreshVariable variable
         , MonadUnify unifier
         )
-    => TermLike variable
+    => Predicate variable
+    -> TermLike variable
     -> TermLike variable
     -> unifier (Predicate variable)
-unificationProcedure  p1 p2
+unificationProcedure predicate p1 p2
```

Comment:
`unificationProcedure` creates a configuration predicate, so we shouldn't need to take one as an argument.